mitiq/zne/inference.py

```diff
@@ -739,10 +771,24 @@ def scale_factor_to_expectation_value(
         ) -> float:
             """Evaluates the quantum expectation value for a given
             scale_factor and other executor parameters."""
-            expectation_values = []
+            expectation_values: List[QuantumResult] = []
```

Comment:
If I am not wrong:
```suggestion
            expectation_values: List[float] = []
```

Reply:
Done (see change below). `executor` returns `QuantumResult`, hence the original. It must return `float` if `observable` is none though, so needed to tell mypy that.